node_modules/col-assets/lib/api.js

```diff
@@ -364,110 +364,116 @@ var getAssets = module.exports.getAssets = function(ctx, filters, sort, limit, o
+    whereClause += hasComments ? ' AND a.comment_count > 0' : ' AND a.comment_count = 0';
   }
 
   var hasImpact = CollabosphereUtil.getBooleanParam(filters.hasImpact, null);
   if (hasImpact !== null) {
-    sqlQuery += hasImpact ? ' AND a.impact_score > 0' : ' AND a.impact_score = 0';
+    whereClause += hasImpact ? ' AND a.impact_score > 0' : ' AND a.impact_score = 0';
   }
 
   var hasLikes = CollabosphereUtil.getBooleanParam(filters.hasLikes, null);
   if (hasLikes !== null) {
-    sqlQuery += hasLikes ? ' AND a.likes > 0' : ' AND a.likes = 0';
-  }
-
-  var hasPins = CollabosphereUtil.getBooleanParam(filters.hasPins, null);
-  if (hasPins !== null) {
-    sqlQuery += hasPins ? ' AND p.asset_id IS NOT NULL' : ' AND p.asset_id IS NULL';
-    if (hasPins && filters.user) {
-      sqlQuery += ' AND p.user_id = :user';
-    }
+    whereClause += hasLikes ? ' AND a.likes > 0' : ' AND a.likes = 0';
   }
 
   var hasTrending = CollabosphereUtil.getBooleanParam(filters.hasTrending, null);
   if (hasTrending !== null) {
-    sqlQuery += hasTrending ? ' AND a.trending_score > 0' : ' AND a.trending_score = 0';
+    whereClause += hasTrending ? ' AND a.trending_score > 0' : ' AND a.trending_score = 0';
   }
 
   var hasViews = CollabosphereUtil.getBooleanParam(filters.hasViews, null);
   if (hasViews !== null) {
-    sqlQuery += hasViews ? ' AND a.views > 0' : ' AND a.views = 0';
+    whereClause += hasViews ? ' AND a.views > 0' : ' AND a.views = 0';
   }
 
   // Query that will be used to get the actual assets
-  var resultsSqlQuery = `SELECT
+  var query = `SELECT
     DISTINCT ON (a.id, a.likes, a.views, a.comment_count, a.impact_score, a.trending_score)
     a.*,
-    act.type AS activity_type
+    act.type AS activity_type,
+    EXISTS (SELECT 1 FROM pinned_user_assets WHERE asset_id = a.id AND user_id = :user_id) as pinned_by_me
   `;
-  resultsSqlQuery += sqlQuery;
 
-  if (sort === 'recent') {
-    resultsSqlQuery += ' ORDER BY a.id DESC';
-  } else if (sort === 'likes') {
-    resultsSqlQuery += ' ORDER BY a.likes DESC, a.id DESC';
+  query += fromClause + ' ' + whereClause;
+
+  if (sort === 'likes') {
+    query += ' ORDER BY a.likes DESC, a.id DESC';
   } else if (sort === 'views') {
-    resultsSqlQuery += ' ORDER BY a.views DESC, a.id DESC';
+    query += ' ORDER BY a.views DESC, a.id DESC';
   } else if (sort === 'comments') {
-    resultsSqlQuery += ' ORDER BY a.comment_count DESC, a.id DESC';
+    query += ' ORDER BY a.comment_count DESC, a.id DESC';
   } else if (sort === 'impact') {
-    resultsSqlQuery += ' ORDER BY a.impact_score DESC, a.id DESC';
+    query += ' ORDER BY a.impact_score DESC, a.id DESC';
   } else if (sort === 'trending') {
-    resultsSqlQuery += ' ORDER BY a.trending_score DESC, a.id DESC';
+    query += ' ORDER BY a.trending_score DESC, a.id DESC';
+  } else if (sort === 'myPinned') {
+    query += ' ORDER BY pinned_by_me DESC, a.id DESC';
+  } else {
+    query += ' ORDER BY a.id DESC';
   }
```

Comment:
Previously, `ORDER BY a.id DESC` was associated with `(sort === 'recent')` with no default ordering. The primary change in diff above is that I made `ORDER BY a.id DESC` the default.